tracing-core/src/lib.rs

```diff
@@ -219,6 +216,14 @@ pub mod span;
 pub(crate) mod stdlib;
 pub mod subscriber;
 
+// Vendored version of spin 0.5.2 (0387621)
+// `mutex` and `once` modules only
+#[cfg(not(feature = "std"))]
+pub mod spin;
```

Comment:
I believe that only the `Once` type from `spin` needs to be public (it's used by the `tracing` macros). Can we make the `spin` mod `pub(crate)`, and only reexport `Once`?

 It should also be `doc(hidden)` — I really want to avoid user code depending on our vendored version of `spin`.

I think this should look more like this:
```rust
#[cfg(not(feature = "std"))]
pub(crate) mod spin;

#[cfg(not(feature = "std"))]
#[doc(hidden)]
pub use self::spin::Once;